spec/spec_helper.rb

```diff
@@ -1,51 +1,39 @@
+# frozen_string_literal: true
+
 require 'pathname'
 dir = Pathname.new(__FILE__).parent
 $LOAD_PATH.unshift(dir, File.join(dir, 'fixtures/modules/augeasproviders_core/spec/lib'), File.join(dir, '..', 'lib'))
 
-require 'rubygems'
+# Managed by modulesync - DO NOT EDIT
+# https://voxpupuli.org/docs/updating-files-managed-with-modulesync/
 
-require 'simplecov'
-unless RUBY_VERSION =~ /^1\.8/
-  require 'coveralls'
-  SimpleCov.formatter = Coveralls::SimpleCov::Formatter
-end
-SimpleCov.start do
-  add_group "Puppet Types", "/lib/puppet/type/"
-  add_group "Puppet Providers", "/lib/puppet/provider/"
+# puppetlabs_spec_helper will set up coverage if the env variable is set.
+# We want to do this if lib exists and it hasn't been explicitly set.
+ENV['COVERAGE'] ||= 'yes' if Dir.exist?(File.expand_path('../lib', __dir__))
 
-  add_filter "/spec/fixtures/"
-  add_filter "/spec/unit/"
-  add_filter "/spec/support/"
+RSpec.configure do |c|
+  c.mock_with :mocha
 end
```

Comment:
It comes from https://github.com/voxpupuli/puppet-augeasproviders_core/blob/4af35652e0f5343178dd824b9e5ce54a3278bd3f/spec/lib/augeas_spec.rb#L17-L19 so I'm not sure if this part actually works. Perhaps it's better to create a patch to augeasproviders_core first to replace it with rspec-mocks. That may be easier and help you in other modules.

Reply:
I just realized (good morning) that _core is actually rspec-mocks already. So it's really the problem that it's mixing things. I'd recommend to rewrite this module to rspec-mocks too.